ui/pages/token-allowance/token-allowance.js

```diff
@@ -50,6 +51,9 @@ import { useGasFeeContext } from '../../contexts/gasFee';
 import { getCustomTxParamsData } from '../confirm-approve/confirm-approve.util';
 import { setCustomTokenAmount } from '../../ducks/app/app';
 import { valuesFor } from '../../helpers/utils/util';
+import { calcTokenAmount } from '../../../shared/lib/transactions-controller-utils';
+
+const MAX_UNSIGNED_256_INT = new BigNumber(2).pow(256).minus(1).toString(10);
```

Comment:
We have now duplicated this BigNumber, can we elevate this to a constant in shared/tokens maybe? I'm assuming this is the maximum allowable value for a token allowance? If thats the case can we do something like: MAX_TOKEN_ALLOWANCE_AMOUNT so that its clearer what the value is?

Reply:
I agree, small generic methods and also constants like `const regex = /^[0-9]{1,}([,.][0-9]{1,})?$/u;` can be moved to shared code.

Reply:
@brad-decker @jpuri Done [here](https://github.com/MetaMask/metamask-extension/pull/16676/commits/0b21618422d67ff022882000efc594ebd62fbd2c).